### PR TITLE
CI: build pdf documentation on circleci

### PR DESCRIPTION
**Pull request by pv**
Build PDF documentation on CircleCI. This shouldn't increase the build time significantly.

This PR also contains some misc. sphinx/latex markup cleanups.

### CODE REVIEW

File: scipy/stats/_distn_infrastructure.py [hunk 1]
```diff
@@ -54,91 +54,91 @@ def instancemethod(func, obj, cls):
...
     (Differential) entropy of the RV.
 """
 _doc_fit = """\
-``fit(data, %(shapes)s, loc=0, scale=1)``
+fit(data, %(shapes)s, loc=0, scale=1)
     Parameter estimates for generic data.
 """
 _doc_expect = """\
-``expect(func, args=(%(shapes_)s), loc=0, scale=1, lb=None, ub=None, conditional=False, **kwds)``
+expect(func, args=(%(shapes_)s), loc=0, scale=1, lb=None, ub=None, conditional=False, **kwds)
```
Comment (ev-br) [inline scipy/stats/_distn_infrastructure.py]: I think these double backticks were there on purpose, to shield asterisks from Sphinx.

Comment (pv) [reply scipy/stats/_distn_infrastructure.py]: They don't seem to be necessary any more --- probably thanks to numpydoc upgrade.

**Comment by pv** [comment]
Outputs:
https://6308-1460385-gh.circle-artifacts.com/0/html-scipyorg/scipy-ref.pdf
https://6308-1460385-gh.circle-artifacts.com/0/html-scipyorg/index.html

**Comment by ev-br** [comment]
https://6308-1460385-gh.circle-artifacts.com/0/html-scipyorg/spatial.html

Rendering of the Distance computations subsection is not right (on a phone screen at least).
Can be fixed separately though

**Comment by pv** [comment]
Right, that's probably due to some autosummary change in sphinx 1.7 and not directly related to this PR.

**Comment by WarrenWeckesser** [comment]
Wow, the PDF document is 2267 pages!  Is this really practical?

The references don't work correctly in the PDF output.  In the docstrings, the "References" section only shows the numbers in brackets, e.g.

**References**

_[1], [2]_

The actual references are in the "Bibliography" section, but the numbers [1], [2], etc., are repeated many times.  It hasn't assigned a unique number to each reference.  E.g.

[1] ‘A Fast Cosine Transform in One and Two Dimensions’, by J. Makhoul, IEEE Transactions on acoustics, speech and signal processing vol. 28(1), pp. 27-34, http://dx.doi.org/10.1109/TASSP.1980.1163351 (1980).

[2] Wikipedia, “Discrete cosine transform”, http://en.wikipedia.org/wiki/Discrete_cosine_transform

[1] Wikipedia, “Discrete sine transform”, http://en.wikipedia.org/wiki/Discrete_sine_transform

[1] ‘Romberg’s method’ http://en.wikipedia.org/wiki/Romberg%27s_method

[1] Wikipedia page: http://en.wikipedia.org/wiki/Trapezoidal_rule

[2] Illustration image: http://en.wikipedia.org/wiki/File:Composite_trapezoidal_rule_illustration.png

[1] J. R. Dormand, P. J. Prince, “A family of embedded Runge-Kutta formulae”, Journal of Computational and Applied Mathematics, Vol. 6, No. 1, pp. 19-26, 1980.

[2] L. W. Shampine, “Some Practical Runge-Kutta Formulas”, Mathematics of Computation„ Vol. 46, No. 173, pp. 135-150, 1986.

**Comment by pv** [comment]
The reference issue is a numpydoc bug (unrelated to this PR which just enables the PDF build on the CI).

We have shipped PDF versions of the docs for all versions of scipy:
https://docs.scipy.org/doc/ indeed they're thousands of pages, and hopefully nobody prints them...

**Comment by pv** [comment]
Previously the PDF files have always been built manually, and this usually turns up issues with invalid latex math markup (which the HTML build tolerates), and those are annoying to fix just before release.

**Comment by ev-br** [comment]
Building the PDF docs on CI looks sensible, as it allows us to actually see this behemoth of a PDF with all the issues it has, so maybe we'll actually look into fixing them. Or decide it's not worth it anymore. Or something else. Meanwhile, let's have it built on CI. So I'll merge this, and we can continue discussing here or in a separate ticket. Thanks Pauli, all.

**Pull request merged**